streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -325,7 +329,7 @@ public void to(Serde<K> keySerde, Serde<V> valSerde, StreamPartitioner<K, V> par
         topology.addProcessor(name, new KStreamTransformValues<>(valueTransformerSupplier), this.name);
         topology.connectProcessorAndStateStores(name, stateStoreNames);
 
-        return new KStreamImpl<>(topology, name, sourceNodes);
+        return new KStreamImpl<>(topology, name, sourceNodes, false);
```

Comment:
This should be new KStreamImpl<>(topology, name, sourceNodes, this.repartitionRequired)`?